account_bank_statement_import_auto_reconcile/models/account_bank_statement_import.py

```diff
@@ -11,37 +11,35 @@ class AccountBankStatementImport(models.TransientModel):
     auto_reconcile = fields.Boolean('Auto reconcile', default=True)
 
     @api.model
-    def _create_bank_statement(self, stmt_vals):
-        statement_id, notifications = super(
+    def _create_bank_statements(self, stmt_vals):
+        statement_ids, notifications = super(
             AccountBankStatementImport, self
-        )._create_bank_statement(stmt_vals)
-        if not statement_id:
-            return statement_id, notifications
-        statement = self.env['account.bank.statement'].browse(statement_id)
-        if (
-                not statement.journal_id
-                .statement_import_auto_reconcile_rule_ids or
-                not self.auto_reconcile
-        ):
-            return statement_id, notifications
-        reconcile_rules = statement.journal_id\
-            .statement_import_auto_reconcile_rule_ids.get_rules()
-        auto_reconciled_ids = []
-        for line in statement.line_ids:
-            for rule in reconcile_rules:
-                if rule.reconcile(line):
-                    auto_reconciled_ids.append(line.id)
-                    break
-        if auto_reconciled_ids:
-            notifications.append({
-                'type': 'warning',
-                'message':
-                _("%d transactions were reconciled automatically.") %
-                len(auto_reconciled_ids),
-                'details': {
-                    'name': _('Automatically reconciled'),
-                    'model': 'account.bank.statement.line',
-                    'ids': auto_reconciled_ids,
-                },
-            })
-        return statement_id, notifications
+        )._create_bank_statements(stmt_vals)
+        if not statement_ids:
+            return statement_ids, notifications
+        statements = self.env['account.bank.statement'].browse(statement_ids)
+        for statement in statements.filtered(
+                lambda x: x.journal_id.
+                statement_import_auto_reconcile_rule_ids or
+                self.auto_reconcile):
```

Comment:
this will unconditionally reconcile as soon as there are rules. But `self.auto_reconcile==False` indicates the user turned off the auto reconcile feature for an import run.
So better return before the loop if `not self.auto_reconcile`, and if you want to filter for truthyness of some dotted path, you can just give it as a string: https://github.com/OCA/OCB/blob/10.0/odoo/models.py#L5027 (costs a tiny bit of performance, but not relevant here and makes the thing more readable)